core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -24,4 +26,31 @@
  */
 public abstract class FileInStream extends InputStream implements BoundedStream, PositionedReadable,
     Seekable {
+    /**
+     * Reads up to len bytes of data from the input stream into the byte buffer.
+     *
+     * @param byteBuffer the buffer into which the data is read
+     * @param off the start offset in the buffer at which the data is written
+     * @param len the maximum number of bytes to read
+     * @return the total number of bytes read into the buffer, or -1 if there is no more data because
+     *         the end of the stream has been reached
+     */
+    public int read(ByteBuffer byteBuffer, int off, int len) throws IOException {
+        int nread = 0;
+        int rd = 0;
+        final int sz = len;
+        final byte[] dest = new byte[sz];
+        while (rd >= 0 && nread < sz) {
```

Comment:
Is this while loop necessary? @LuQQiu @apc999 , this code fragment is copied from `AlluxioFuseFileSystem#copyInternal`.